private/buf/buflsp/file.go

```diff
@@ -371,6 +372,24 @@ func (f *file) FindModule(ctx context.Context) {
 		return
 	}
 
+	// Get the check runner provider for this file. The client is scoped to
+	// the inputs buf.lock file, so we need to get the check runner provider
+	// for the workspace that contains this file.
```

Comment:
The LSP is currently supported but in testing it can result in a large number of queued checks causing the LSP state to lag behind the editor. It looks like each file change causes a lint command to be executed. As each lint is slower then the number of changes in the file many false annotations are provided before the current file state is reached. This is exaggerated with Wasm plugins as they are slower then native.

Reply:
Fixed in https://github.com/bufbuild/buf/pull/3547